### PR TITLE
STM32 DCC Decoder modules

### PR DESCRIPTION
**Pull request by balazsracz**
Adds a hardware module for the STM32 DCC decoder.
Adds an example app to decode DCC signal on a nucleo STM32F091.

### CODE REVIEW

File: applications/dcc_decoder/targets/freertos.armv6m.st-stm32f091rc-nucleo/HwInit.cxx [hunk 1]
```diff
@@ -0,0 +1,327 @@
+/** \copyright
+ * Copyright (c) 2018, Balazs Racz
```
Comment (bakerstu) [inline applications/dcc_decoder/targets/freertos.armv6m.st-stm32f091rc-nucleo/HwInit.cxx:2]: date

File: applications/dcc_decoder/targets/freertos.armv6m.st-stm32f091rc-nucleo/HwInit.cxx [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * \file HwInit.cxx
+ * This file represents the hardware initialization for the STM32F091RC Nucelo
+ * board (bare).
+ *
+ * @author Balazs Racz
+ * @date April 18, 2018
```
Comment (bakerstu) [inline applications/dcc_decoder/targets/freertos.armv6m.st-stm32f091rc-nucleo/HwInit.cxx:32]: date

File: src/freertos_drivers/st/Stm32DCCDecoder.hxx [hunk 3]
```diff
@@ -24,64 +24,95 @@
...
-
-    static const auto OS_INTERRUPT = INT_WTIMER4B;
-    DECL_PIN(NRZPIN, D, 4);
-    static const auto NRZPIN_CONFIG = GPIO_PD4_WT4CCP0;
-
-    static const uint32_t TIMER_MAX_VALUE = 0x8000000UL;
-
-    static const int Q_SIZE = 16;
-
+    /// The Pin (declared as GPIO input) where the DCC signal comes in.
```
Comment (bakerstu) [inline src/freertos_drivers/st/Stm32DCCDecoder.hxx:54]: Very good documentation. I guess old dogs can learn new tricks :)

File: applications/dcc_decoder/targets/freertos.armv6m.st-stm32f091rc-nucleo/HwInit.cxx (hunk 1 above)
Comment (balazsracz) [reply applications/dcc_decoder/targets/freertos.armv6m.st-stm32f091rc-nucleo/HwInit.cxx:2]: done

File: applications/dcc_decoder/targets/freertos.armv6m.st-stm32f091rc-nucleo/HwInit.cxx (hunk 2 above)
Comment (balazsracz) [reply applications/dcc_decoder/targets/freertos.armv6m.st-stm32f091rc-nucleo/HwInit.cxx:32]: done

File: src/freertos_drivers/st/Stm32DCCDecoder.hxx (hunk 3 above)
Comment (balazsracz) [reply src/freertos_drivers/st/Stm32DCCDecoder.hxx:54]: thanks

**Pull request merged**